go/runtime/host/provisioner/provisioner.go

```diff
@@ -34,6 +35,7 @@ func New(
 	commonStore *persistent.CommonStore,
 	identity *identity.Identity,
 	consensus consensus.Service,
+	genesisDoc *genesisAPI.Document,
```

Comment:
Could it instead get the genesis document from `consensus`?

Reply:
Yes, but it is not as efficient and we would be calling `consensus` during initialization of structs, which is not the best, as consensus might not be up (p2p peers are definitely not available at this stage). A better solution would call `consensus` once on startup when things are already running.

Reply:
An alternative is to create custom functions for these values, but again, we would be calling endpoints from constructors.